src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurSaeCreateRequestedCashAdvanceFileValidationServiceImpl.java

```diff
@@ -0,0 +1,398 @@
+import edu.cornell.kfs.concur.batch.service.ConcurRequestedCashAdvanceService;
+import edu.cornell.kfs.concur.batch.service.ConcurSaeCreateRequestedCashAdvanceFileValidationService;
+import edu.cornell.kfs.concur.batch.service.ConcurStandardAccountingExtractCashAdvanceService;
+import edu.cornell.kfs.concur.businessobjects.ConcurAccountInfo;
+import edu.cornell.kfs.concur.businessobjects.ValidationResult;
+import edu.cornell.kfs.concur.service.ConcurAccountValidationService;
+
+public class ConcurSaeCreateRequestedCashAdvanceFileValidationServiceImpl implements ConcurSaeCreateRequestedCashAdvanceFileValidationService {
+    private static final Logger LOG = LogManager.getLogger(ConcurSaeCreateRequestedCashAdvanceFileValidationServiceImpl.class);
+    protected ConcurRequestedCashAdvanceService concurRequestedCashAdvanceService;
+    protected ConcurAccountValidationService concurAccountValidationService;
+    protected ConfigurationService configurationService;
+    protected ConcurBatchUtilityService concurBatchUtilityService;
+    protected ConcurEmployeeInfoValidationService concurEmployeeInfoValidationService;
+    protected ConcurStandardAccountingExtractCashAdvanceService concurStandardAccountingExtractCashAdvanceService;
+
+    @Override
+    public boolean saeHeaderRowValidatesToFileContentsForRequestedCashAdvances(ConcurStandardAccountingExtractFile standardAccountingExtractFile, ConcurSaeRequestedCashAdvanceBatchReportData reportData) {
+        boolean headerValidationPassed;
+        LOG.info("saeHeaderRowValidatesToFileContentsForRequestedCashAdvances:  batchDate=" + standardAccountingExtractFile.getBatchDate() + "=   recordCount=" + standardAccountingExtractFile.getRecordCount() + "=     journalAmountTotal=" + standardAccountingExtractFile.getJournalAmountTotal() + "=");
+        headerValidationPassed = (fileRowCountMatchesHeaderRowCount(standardAccountingExtractFile, reportData) &&
+                                  fileOnlyContainsOurEmployeeCustomerIndicator(standardAccountingExtractFile, reportData) &&
+                                  fileHeaderJournalTotalMatchesFileDetailsCalculatedJournalAmountTotal(standardAccountingExtractFile, reportData));
+        LOG.info("saeHeaderRowValidatesToFileContentsForRequestedCashAdvances:  Header validation : " + ((headerValidationPassed) ? "PASSED" : "FAILED"));
+        return headerValidationPassed;
+    }
+
+    @Override
+    public void performSaeDetailLineValidationForRequestedCashAdvances(ConcurStandardAccountingExtractDetailLine detailFileLine, List<String> uniqueRequestedCashAdvanceKeysInFile) {
+        if (requestDetailFileLineIsCashAdvance(detailFileLine)) {
+            boolean lineValidationPassed = true;
+            lineValidationPassed =  requestedCashAdvanceApprovedByConcurAdministrator(detailFileLine);
+            LOG.info("requestedCashAdvanceApprovedByConcurAdministrator = " + lineValidationPassed);
```

Comment:
the function name should be included in these log statements

Reply:
Function name added.